neard/src/cli.rs

```diff
@@ -32,11 +34,12 @@ impl NeardCmd {
     pub(super) fn parse_and_run() -> Result<(), RunError> {
         let neard_cmd = Self::parse();
 
-        let runtime = tokio::runtime::Runtime::new().unwrap();
-        // Opentelemetry needs a running Tokio system to report spans asynchronously and in batches,
-        // which is great for the overall system performance.
-        let _subscriber_guard =
-            runtime.block_on(async { init_logging(&neard_cmd.opts).await.unwrap() });
+        // Enable logging of the current thread.
+        let _subscriber_guard = default_subscriber(
+            make_env_filter(neard_cmd.opts.verbose_target())?,
+            &neard_cmd.opts.o11y,
+        )
+        .local();
```

Comment:
An approach I had used in the past was instead:

1. Save all the events, spans into a some sort of buffer;
2. Once the configuration is loaded, initialize the logging system;
3. Replay all of the saved events, spans again, now that the subscriber has been initialized.

While the current approach works for now, it is probably going to be quite inconvenient if there's a need to track down some consequential information being emit during the prep stage.

Reply:
I actually like the current approach, because it:
1) lets us easily and reliably configure logging for simple uses cases, such as tools/storage-usage-delta-calculator/src/main.rs . Previously those use cases created tokio runtime and then initialized logging. I remember seeing that one of the tools was not able to log anything at all because of mis-managing that tokio runtime, but I didn't have time to fix it.
2) Logging from the `main()`-thread directly is good enough and unlikely that we need non-blocking logging from that thread. Ideally a node should be up and running after a restart as soon as possible, which means it is supposed to do as little work as possible in the `main()`-thread.
3) Maybe I don't understand how buffering is supposed to work, but here is my understanding. The `main()`-thread buffers the spans and logs until nearconfig is available, which means anything logged before loading nearconfig will only be shown to the user after nearconfig is loaded. For example, if an error is encountered when reading nearconfig, that error will not be able to be shown to the user.

Reply:
I think that’s fine, yeah. The only thing I’m not seeing is this subscriber guard being dropped sometime around the point in time where the global subscriber is initialized.

This, I believe, would mean that the main thread uses the simplified logger indefinitely, which doesn’t seem quite right to me.

Reply:
Well, once there is a problem, being able to figure out what version a broken node was running is useful, isn’t it?

Though, again, this is kinda back to the discussion about whether we consider needing to hunt down and look at the local logs for the particular broken instance. I personally don’t think that’s super convenient, and would rather if everything eventually made into the same place (for regular neard validator deployments, anyway.)

It isn’t necessarily the case that this needs to be moved per se, I could see this being solved by duplication (i.e. log the version _again_ after initializing the otel logger.)

The caveat is that I’m not an operating a validator, so all I’m saying is basically just me armchairing.